setuptools/tests/test_editable_install.py

```diff
@@ -577,6 +577,27 @@ def test_similar_name(self, tmp_path):
             with pytest.raises(ImportError, match="foobar"):
                 import_module("foobar")
 
+    def test_case_sensitivity(self, tmp_path):
+        files = {
+            "foo": {
+                "__init__.py": "",
+                "lowercase.py": "x = 1",
+            },
+        }
```

Comment:
What would happen here if I have a file structure like

```python
files = {
    "foo": {
        "__init__.py": "",
        "bar": {
            "__init__.py": "",
            "lowercase.py": "x = 1",
        }
    }
}
```

and I try to import `foo.BAR.lowercase`?

Reply:
Shoot, you're right this fails. I will look into an implementation with `importlib.machinery` as mentioned below.